core/opendaq/reader/src/block_reader_impl.cpp

```diff
@@ -163,27 +196,56 @@ ErrCode BlockReaderImpl::readPacketData()
         }
     }
 
-    if (toRead < remainingSampleCount)
+    info.writtenSampleCount += sampleCountToRead;
+    info.prevSampleIndex += sampleCountToRead;
+    info.remainingSamplesToRead -= sampleCountToRead;
+
+    if (info.writtenSampleCount % blockSize == 0)
     {
-        info.prevSampleIndex += toRead;
+        // how many data packet need to be stepped back
+        const auto rewindSamples = overlappedBlockSize;
```

Comment:
I would refactor this section for calculating how many packets/samples to rewind into a separate function that's called only if overlap > 0 for the sake of code clarity.

Reply:
Done